optaplanner-wb-screens/optaplanner-wb-domain-editor/optaplanner-wb-domain-editor-backend/src/main/java/org/optaplanner/workbench/screens/domaineditor/backend/server/PlannerDomainHandler.java

```diff
@@ -66,15 +66,17 @@ public void setDefaultValues( DataObject dataObject, Map<String, Object> options
 
     @Override
     public void processDataObject( DataObject dataObject, DataModel dataModel ) {
-        boolean hasPlanningEntity = dataObject.getAnnotation( PlannerDomainAnnotations.PLANNING_ENTITY_ANNOTATION ) != null;
-        boolean hasComparatorDefinition = dataObject.getAnnotation( ComparatorDefinition.class.getName() ) != null;
-        if ( hasPlanningEntity && hasComparatorDefinition ) {
-            try {
-                ComparatorObject comparatorObject = plannerDataObjectEditorService.extractComparatorObject( dataObject, dataModel );
-                dataObject.getNestedClasses().clear();
-                dataObject.getNestedClasses().add( comparatorObject );
-            } catch ( Exception e ) {
-                logger.error( "Unable to extract comparator object from " + dataObject.getClassName() );
+        if ( dataModel != null ) {
+            boolean hasPlanningEntity = dataObject.getAnnotation( PlannerDomainAnnotations.PLANNING_ENTITY_ANNOTATION ) != null;
+            boolean hasComparatorDefinition = dataObject.getAnnotation( ComparatorDefinition.class.getName() ) != null;
+            if ( hasPlanningEntity && hasComparatorDefinition ) {
+                try {
+                    ComparatorObject comparatorObject = plannerDataObjectEditorService.extractComparatorObject( dataObject, dataModel );
+                    dataObject.getNestedClasses().clear();
+                    dataObject.getNestedClasses().add( comparatorObject );
+                } catch ( Exception e ) {
```

Comment:
Clarity note: I am not advocating to add more "java checked exceptions", usually a subclass of RuntimeException (such as IllegalStateException) does the trick. Note that I mean "java checked exception" as in java forcing you to write a catch, for example "throws IOException".